frontend/server/src/main/java/org/pytorch/serve/wlm/Model.java

```diff
@@ -281,7 +283,13 @@ public void pollBatch(String threadId, long waitTime, Map<String, Job> jobsRepo)
                 }
                 maxDelay -= end - begin;
                 begin = end;
-                jobsRepo.put(j.getJobId(), j);
+                if (j.getPayload().getClientExpireTS() > System.currentTimeMillis()) {
+                    jobsRepo.put(j.getJobId(), j);
```

Comment:
Is the client timeout w.r.t to a single inference example or a single batch?

Reply:
timeout is applied on each inference request.